public/app/plugins/datasource/influxdb/fsql/sqlUtil.ts

```diff
@@ -0,0 +1,45 @@
+import { isEmpty } from 'lodash';
+
+import { SQLQuery } from 'app/features/plugins/sql/types';
+import { createSelectClause, haveColumns } from 'app/features/plugins/sql/utils/sql.utils';
+
+export function toRawSql({ sql, dataset, table }: SQLQuery): string {
+  let rawQuery = '';
+
+  // Return early with empty string if there is no sql column
+  if (!sql || !haveColumns(sql.columns)) {
+    return rawQuery;
+  }
+
+  rawQuery += createSelectClause(sql.columns);
+
+  if (dataset && table) {
+    rawQuery += `FROM ${dataset}.${table} `;
+  }
+
+  if (sql.whereString) {
+    rawQuery += `WHERE ${sql.whereString} `;
+  }
+
+  if (sql.groupBy?.[0]?.property.name) {
```

Comment:
do we want to iterate through the groupBy or are we fine selecting the first one?

Reply:
This code was taken from mysql and postgresql datasources. For now it is ok but it needs some refactoring to fit with InfluxDB SQL requirements. Code generation is a bit problematic and I will address that in another PR. And I will definitely check that part too.